Algorithms/Encoders/NYSIISEncoder.cs

```diff
@@ -0,0 +1,197 @@
+                });
+
+                //H[vowel] or [vowel]H -> text[i-1]
+                if(text[i] == 'H')
+                {
+                    if(Vowels.Contains(text[i - 1]))
+                        text = text.ReplaceSingle(i, text[i - 1]);
+                    else if(i < text.Length - 1 && Vowels.Contains(text[i + 1]))
+                        text = text.ReplaceSingle(i, text[i - 1]);
+                }
+
+                //[vowel]W -> [vowel]
+                if(text[i] == 'W' && Vowels.Contains(text[i - 1]))
+                    text = text.ReplaceSingle(i, text[i - 1]);
+
+                if(nysiis.Last() != text[i] && text[i] != ' ' && char.IsUpper(text[i]))
+                    nysiis += text[i];
+            }
+
+            //ends in S => remove
+            if(nysiis.Last() == 'S') nysiis = nysiis.Substring(0, nysiis.Length - 1);
+            //ends in AY => replace with single Y
+            if(nysiis.Last() == 'Y' && nysiis[nysiis.Length - 2] == 'A')
+            {
+                nysiis = nysiis.ReplaceAt(nysiis.Length - 2, "Y", 2);
+            }
+            //ends in A => remove
+            if(nysiis.Last() == 'A') nysiis = nysiis.Substring(0, nysiis.Length - 1);
+
+            return nysiis;
+        }
+
+        private string RepWith(string start, int index, (char, char)[] opts)
+        {
+            for(int i = 0; i < opts.Length; i++)
+            {
+                if(start[index] == opts[i].Item1) return start.ReplaceSingle(index, opts[i].Item2);
+            }
+            return start;
+        }
+
+        private string RepArr(string start, int index, (string, string)[] opts)
+        {
+            for(int i = 0; i < opts.Length; i++)
+            {
+                string check = opts[i].Item1;
+                string repl = opts[i].Item2;
+                if(start.Length >= index + check.Length && start.Substring(index, check.Length) == check)
+                    return start.ReplaceAt(index, repl, check.Length);
+            }
+            return start;
+        }
+
+        private string StartReplace(string start, (string, string)[] checks)
+        {
+            checks.ForEach(tup => {
+                string pref = tup.Item1;
+                string rep = tup.Item2;
+                if(start.StartsWith(pref))
+                    start = start.ReplaceAt(0, rep, pref.Length);
+            });
+            return start;
+        }
+
+        private string EndReplace(string end, (string, string)[] checks)
+        {
+            checks.ForEach(tup => {
+                string suff = tup.Item1;
+                string rep = tup.Item2;
+                if(end.EndsWith(suff))
+                    end = end.ReplaceAt(end.Length - suff.Length, rep, suff.Length);
+            });
+            return end;
+        }
+    }
+
+    /// <summary>
+    /// Static class containing Extension Methods used by the NYSIIS algorithm.
+    /// </summary>
+    public static class NYSIISExtensions
```

Comment:
Codacy found an issue: [Rename class 'NYSIISExtensions' to match pascal case naming rules, consider using 'NysiisExtensions'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6311443)